src/components/Game/Timer/Timer.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import classes from './Timer.scss';
+import { computeMinutes, computeSeconds, printTwoDigits } from '../../../helpers';
+
+class Timer extends Component {
+    state = {
+        // 60 sec. only for testing reasons. it's gonna be more :)
+        seconds: 60,
+    };
+
+    componentDidMount() {
+        this.startTimer();
+    }
+
+    componentWillUnmount() {
+        this.stopTimer();
+    }
+
+    startTimer = () => {
+        const { timeIsOver } = this.props;
+
+        this.interval = setInterval(() => {
+            const { seconds } = this.state;
+
+            this.decrement();
+            if (seconds <= 1) {
```

Comment:
about moving up: when I do this I get such an error
> Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.

the condition is `<=1` because the timer shouldn't show 0:00 (as it shows 1:00). otherwise a player would have one extra second to play. 

> argument: hypothetically you may want to change interval value (now set at 1000ms) and in current solution you may need to update two places, now value 1 in the above line and value 1000 on line 32 are logically the same values, having the same piece of information in two places is something we should avoid

it's not very clear to me what your proposal is in here 😁 defining a constant?

Reply:
1. The condition for checking if time has run out should be in the shape of `if (timeLeft <= 0) {…`.
2. Please have a look at lines 28, 32 and 37. All these lines includes the same piece of information: namely the fact the interval is 1 second long. It's preferable when a single piece of information is defined once.
3. I would expect (and that's what I have seen in timers in recruitment tools) that the 60-second timer immediately jumps to 0:59 (because just after time starts it's 59.99 seconds so 59 after all), then 0:58, …, 0:01, 0:00 (displayed for 1 second as it's 0.99, 0.98, 0.97, …) -> "bang, time ends!"

Reply:
👍